### ISSUE TITLE
Atmos Badge

### ISSUE THREAD

**Issue by jaydecay78** [RatingPosterDB/rpdb-folders#48]
Since the last update the Atmos badge isn't pulling through onto the final poster - shows correctly in the preview but is missing from file that is downloaded

**Comment by jaydecay78** [comment]
actually - ignore me - is now working fine - very odd

**Comment by jaruba** [comment]
@jaydecay78 there might actually be an issue with that as this is the second report of this type that we received since the latest release, i will look into it

**Comment by jaruba** [comment]
@jaydecay78 can you tell me if you are using radarr / sonarr webhooks or FS events?

I have re-opened this issue as there is definitely some badging issue.

**Comment by jaydecay78** [comment]
Not using wither of those :)

**Comment by jaruba** [comment]
@jaydecay78 There are only 3 options in "Scan Settings":
- FS Events (default)
- Polling
- Radarr / Sonarr WebHooks

If you never changed these settings, than it should use the default "FS Events".

Can you also check the folder of the movie or series that this happened with? Do you by any chance have 2 posters and 2 backdrops? Instead of one poster and one backdrop?

**Comment by Oizopower** [comment]
For me it's not working either.

Using the Docker version.
- Radarr / Sonarr WebHooks

I see in the logfile / output the correct series / movies being called.
But when i look in the folder, there is no modified poster file. Once i do "Edit Movie/Series poster" and update it, it works.

**Comment by BlueBull010** [comment]
I'm having the same issue at the moment. No badges are added anymore. Three strange things: Firstly, if I manually edit a movie and add badges with the exact same settings as the automatic mode, badges are in fact added. Secondly: it's only for my movies folder that it doesn't add badges anymore, series folder isn't affected. Any idea what I could do? I'm willing to provide logs or anything else that can help if you give me instructions. And lastly: The container has been on the newest version for months now. No update was done and so it used to work with this version, it just stopped working without doing an update or anything

Things I tried:
- Restart container
- Checked requests left (waaaaay more left than I need)
- Removed badge settings from the folder and set them exactly the same again
- Do a full overwrite scan
- Manually remove all posters and rpdb.json files and then do a full rescan

I find it bizarre that it doesn't work for my movie folder but does work for my series folder

**Comment by jaruba** [comment]
@BlueBull010 that is very strange, maybe it is an issue with persisting the settings for the docker container? as 2/3 users that had this issue are using docker, and the 3rd did not mention how he uses RPDB (so he may be using docker too)

the way we could prove that it could be a settings persistance issue, is by restarting the docker container to see if that breaks it again

**Comment by BlueBull010** [comment]
@jaruba Thanks for your reply. The container has been restarted multiple times since my last message and it seemingly still works. Would it be possible for setting persistence to only affect very specific settings in stead of all of them? Because changes I make to all settings do seem to persist, but I'm not a developer so it's quite possible I misunderstand what exactly you mean by that*

I have, however, noticed that manual matching of unmatched media has never worked for me. It does not seem to "persist" my choice in that it just returns to the previous menu without matching it. Same for the "choose poster", "custom poster" and "create poster" functions, they don't actually let me specify a poster (through an image URL, a TMDB page,... or through the "choose poster" menu) they just return to the previous menu and keep saying "missing poster". This is rare, only a fraction of my media has a missing poster in RPDB but in all those cases, choosing a poster or matching a series manually using IMDB URL or IMDB ID does not seem to persist. No idea if this is related but because you mentioned "persistence" and the choosing of a poster and the matching with IMDB do not seem to "persist" either, I thought it maybe worth mentioning

**Comment by BlueBull010** [comment]
I have the same issue again. This time with my folder for TV shows in stead of the movies folder. And this time, also, the previously mentioned workaround does not work. It just doesn't add badges to any TV shows anymore. Could you please assist? It does complain that it can't find any video files to probe in the TV show folder, which is untrue because the video files are there and they are in the same format as my movies. Badges are, next to IMDB ratings, the primary reason that I use (and pay for) this software so I really need to get this fixed

**Comment by jaruba** [comment]
@BlueBull010 can you contact me through the "Contact" button on the website: https://ratingposterdb.com/

please include the email that you used on Patreon when subscribing and any logs you have from the RPDB Folders app so i can investigate

**Comment by BlueBull010** [comment]
> @BlueBull010 can you contact me through the "Contact" button at the bottom of the website: https://ratingposterdb.com/
> 
> please include the email that you used on Patreon when subscribing and any logs you have from the RPDB Folders app so i can investigate

Of course. I will collect them today or tomorrow and send them to you. I will also send you my settings file. I wonder whether this is an issue with RPDB or with something external to RPDB. It's strange that it just starts occurring out of the blue, even stranger that the same issue happens both on docker and windows and even stranger yet that my movies still get badges even though the media files are in the same storage location, with identical file permissions and identical RPDB settings for both movies and TV shows. Because it complains it can't find a media file to probe, it almost seems like something is interfering with the probing service or something

**Comment by BlueBull010** [comment]
TL;DR: Make sure the very first subfolder in a TV show folder, contains video files and not other data. RPDB probably only scans the first subfolder

Alright, so good news. I have found the cause, at least in my case. I use a separate program called "TinyMediaManager", which is used to download fanart, banners, posters, TV show and movie info,... and lots of other metadata. Among other things, it creates a subfolder called ".actors" within every TV show folder, containing images of actors who play in that TV show. Deleting these .actors folders has solved it and badges are now added once more

I suspect that RPDB only scans the first subfolder for video files and since ".actors" is alphabetically the first folder, it doesn't find anything and skips the TV show entirely. Each TV show folder, in my case, contains a subfolder for each season ("Season 1", "Season 2",...) which contains the episode video files, but those were not alphabetically first. This is confirmed by the fact that some TV shows I have only downloaded the latest seasons, so there is for instance an empty "Season 1" folder in there, and those TV shows fail as well, even after I have deleted the ".actors" folder

**Comment by jaruba** [comment]
i understand the issue now, and it should definitely be fixable, i don't believe this has ever been reported before which makes it an edge case, i will revisit the tv show scanning logic with the new details you shared

**Comment by BlueBull010** [comment]
> i understand the issue now, and it should definitely be fixable, i don't believe this has ever been reported before which makes it an edge case, i will revisit the tv show scanning logic with the new details you shared

Awesome, thank you. No rush, as you said it's an edge case and I have a solution now so the issue is gone. Let me know if you need anything from me to fix it, I will gladly assist
